packages/arb-avm-cpp/data_storage/src/arbcore.cpp

```diff
@@ -3714,11 +3715,9 @@ bool ArbCore::logsCursorConfirmReceived(size_t cursor_index) {
         logs_cursors[cursor_index].reorg_mutex);
 
     if (logs_cursors[cursor_index].status != DataCursor::DELIVERED) {
-        logs_cursors[cursor_index].error_string =
-            "logsCursorConfirmReceived called at wrong state";
         std::cerr << "logsCursorConfirmReceived called at wrong state: "
                   << logs_cursors[cursor_index].status << "\n";
-        logs_cursors[cursor_index].status = DataCursor::ERROR;
+        abortThread();
```

Comment:
Isn't this just an API usage error that should be recoverable? Not sure we want to abort here vs just returning an error in the API

Reply:
seems reasonable

Reply:
Leaving it as is.  This should never happen as long as the API is followed, and it isn't always clear what the safest way to clear the error would be.